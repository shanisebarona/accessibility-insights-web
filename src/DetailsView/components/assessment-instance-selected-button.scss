// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';

.instance-visibility-button {
    color: $primary-text;
    padding-bottom: 6px;
    @media screen and (forced-colors: active) {
        color: highlighttext !important;
    }
}

.instance-visibility-button:focus {
    @media screen and (forced-colors: active) {
        outline: 1px highlighttext solid;
    }
}

.test-instance-selected {
    padding: 2px;
    color: $neutral-0;
}

.test-instance-selected-hidden {
    background-color: $neutral-60;
    cursor: default;
}

.test-instance-selected-visible {
    background-color: $communication-primary;
}

.test-instance-selected-hidden-button {
    background-color: transparent;
}
