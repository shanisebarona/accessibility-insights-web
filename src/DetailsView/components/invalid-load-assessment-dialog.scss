// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
.invalid-load-assessment-dialog {
    button {
        width: 140px;
    }
}
