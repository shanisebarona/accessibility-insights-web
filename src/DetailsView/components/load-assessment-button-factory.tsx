// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    LoadAssessmentButton,
    LoadAssessmentButtonProps,
} from 'DetailsView/components/load-assessment-button';
import * as React from 'react';

export function getLoadButtonForAssessment(props: LoadAssessmentButtonProps): JSX.Element {
    return <LoadAssessmentButton {...props} />;
}

export function getLoadButtonForFastPass(props: LoadAssessmentButtonProps): JSX.Element | null {
    return null;
}
