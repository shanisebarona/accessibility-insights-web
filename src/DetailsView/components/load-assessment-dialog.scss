// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

.load-button {
    button {
        width: 140px;
    }
}
