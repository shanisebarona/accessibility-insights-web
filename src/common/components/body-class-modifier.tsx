// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { DocumentManipulator } from 'common/document-manipulator';
import * as React from 'react';

export type BodyClassModifierDeps = {
    documentManipulator: DocumentManipulator;
};

export type BodyClassModifierProps = {
    classNames: string[];
    deps: BodyClassModifierDeps;
};

// We use our own BodyClassModifier rather than using Helmet's similar functionality because we
// need multiple BodyClassModifiers to be able to co-exist without overwriting each other or the
// body classes office fabric maintains.
export class BodyClassModifier extends React.Component<BodyClassModifierProps> {
    public render(): JSX.Element | null {
        return null;
    }

    public componentDidMount(): void {
        this.updateClassNames([], this.props.classNames);
    }

    public componentDidUpdate(prevProps: BodyClassModifierProps): void {
        this.updateClassNames(prevProps.classNames, this.props.classNames);
    }

    public componentWillUnmount(): void {
        this.updateClassNames(this.props.classNames, []);
    }

    private updateClassNames(oldClassNames: string[], newClassNames: string[]): void {
        const { documentManipulator } = this.props.deps;

        const addedClassNames = newClassNames.filter(newName => !oldClassNames.includes(newName));
        const removedClassNames = oldClassNames.filter(oldName => !newClassNames.includes(oldName));

        // document.body may include classes not managed by this BodyClassModifier; leave them as-is
        let allBodyClassNames = documentManipulator.getBodyClassNames();
        allBodyClassNames = allBodyClassNames.filter(n => !removedClassNames.includes(n));
        allBodyClassNames.push(...addedClassNames);

        documentManipulator.setBodyClassNames(allBodyClassNames);
    }
}
