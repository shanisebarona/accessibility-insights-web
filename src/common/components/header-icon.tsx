// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { BrandBlue } from '../../icons/brand/blue/brand-blue';
import { BrandWhite } from '../../icons/brand/white/brand-white';
import { NamedFC } from '../react/named-fc';
import { ThemeInnerState } from './theme';
import {
    withStoreSubscription,
    WithStoreSubscriptionDeps,
    WithStoreSubscriptionProps,
} from './with-store-subscription';

export type HeaderIconState = ThemeInnerState;

export type HeaderIconDeps = WithStoreSubscriptionDeps<HeaderIconState>;

export type HeaderIconProps = {
    invertColors?: boolean;
} & WithStoreSubscriptionProps<HeaderIconState>;

export const HeaderIconComponent = NamedFC<HeaderIconProps>('HeaderIconComponent', props => {
    const state = props.storeState.userConfigurationStoreData;
    const enableHighContrast = state && state.enableHighContrast;

    const { invertColors = false } = props;

    const defaultIcon = invertColors ? <BrandBlue /> : <BrandWhite />;
    const highContrastIcon = invertColors ? <BrandWhite /> : <BrandBlue />;

    return enableHighContrast ? highContrastIcon : defaultIcon;
});

export const HeaderIcon =
    withStoreSubscription<HeaderIconProps, HeaderIconState>(HeaderIconComponent);
