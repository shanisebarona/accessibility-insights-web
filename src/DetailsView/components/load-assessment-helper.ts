// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentDataParser } from 'common/assessment-data-parser';
import { Tab } from 'common/itab';
import { VersionedAssessmentData } from 'common/types/versioned-assessment-data';
import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
import { LoadAssessmentDataValidator } from 'DetailsView/components/load-assessment-data-validator';

export class LoadAssessmentHelper {
    constructor(
        private readonly assessmentDataParser: AssessmentDataParser,
        private readonly detailsViewActionMessageCreator: DetailsViewActionMessageCreator,
        private readonly fileReader: FileReader,
        private readonly document: Document,
        private readonly loadAssessmentDataValidator: LoadAssessmentDataValidator,
    ) {}

    public getAssessmentForLoad(
        setAssessmentState: (versionedAssessmentData: VersionedAssessmentData) => void,
        toggleLoadAssessmentDialog: () => void,
        prevTargetPageData: Tab,
        newTargetPageId: number,
    ): void {
        const input = this.document.createElement('input');
        input.type = 'file';
        input.accept = '.a11ywebassessment';

        const onReaderLoad = (readerEvent: ProgressEvent<FileReader>) => {
            const content = readerEvent.target.result as string;
            let parsedAssessmentData: VersionedAssessmentData;

            try {
                parsedAssessmentData = this.assessmentDataParser.parseAssessmentData(content);
            } catch {
                console.log('Invalid JSON');
                //toggle invalid data dialog
                return;
            }

            const validationData =
                this.loadAssessmentDataValidator.uploadedDataIsValid(parsedAssessmentData);

            if (!validationData.dataIsValid) {
                console.log('Invalid JSON schema');
                console.log(validationData.errors);
                //toggle invalid data dialog
                return;
            }

            setAssessmentState(parsedAssessmentData);

            if (prevTargetPageData != null) {
                toggleLoadAssessmentDialog();
            } else {
                this.detailsViewActionMessageCreator.loadAssessment(
                    parsedAssessmentData,
                    newTargetPageId,
                );
            }
        };
        const onInputChange = (e: Event) => {
            const file = (e.target as HTMLInputElement).files[0];
            this.fileReader.onload = onReaderLoad;
            this.fileReader.readAsText(file, 'UTF-8');
        };

        input.onchange = onInputChange;
        input.click();
    }
}
