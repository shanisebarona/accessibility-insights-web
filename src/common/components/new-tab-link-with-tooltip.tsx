// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NewTabLink } from 'common/components/new-tab-link';
import { ICalloutProps, ILinkProps, ITooltipHostStyles, TooltipHost } from 'office-ui-fabric-react';
import * as React from 'react';
import { NamedFC } from '../react/named-fc';
import * as styles from './new-tab-link-with-tooltip.scss';

export type NewTabLinkWithTooltipProps = ILinkProps & { tooltipContent: string | undefined };

export const NewTabLinkWithTooltip = NamedFC<NewTabLinkWithTooltipProps>(
    'NewTabLinkWithTooltip',
    props => {
        const { tooltipContent, ...linkProps } = props;
        const hostStyles: Partial<ITooltipHostStyles> = {
            root: styles.insightsTooltipHost,
        };
        const calloutProps: ICalloutProps = {
            styles: {
                root: styles.insightsTooltip,
                beak: styles.insightsTooltip,
            },
        };
        return (
            <TooltipHost content={tooltipContent} styles={hostStyles} calloutProps={calloutProps}>
                <NewTabLink {...linkProps} />
            </TooltipHost>
        );
    },
);
