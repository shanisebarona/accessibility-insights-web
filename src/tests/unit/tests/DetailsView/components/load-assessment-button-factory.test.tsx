// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { LoadAssessmentButtonProps } from 'DetailsView/components/load-assessment-button';
import {
    getLoadButtonForAssessment,
    getLoadButtonForFastPass,
} from 'DetailsView/components/load-assessment-button-factory';

describe('LoadAssessmentFactory', () => {
    let props: LoadAssessmentButtonProps;

    describe('getLoadButtonForAssessment', () => {
        test('renders load assessment button', () => {
            const rendered = getLoadButtonForAssessment(props);
            expect(rendered).toMatchSnapshot();
        });
    });

    describe('getLoadButtonForFastPass', () => {
        test('renders load assessment button as null', () => {
            const rendered = getLoadButtonForFastPass(props);
            expect(rendered).toBeNull();
        });
    });
});
