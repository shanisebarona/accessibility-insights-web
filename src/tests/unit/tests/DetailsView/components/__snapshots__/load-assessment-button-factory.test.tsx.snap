// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadAssessmentFactory getLoadButtonForAssessment renders load assessment button 1`] = `<LoadAssessmentButton />`;
